ui/pages/token-allowance/token-allowance.js

```diff
@@ -442,6 +454,57 @@ export default function TokenAllowance({
           />
         </Box>
       )}
+      {useNonceField && (
+        <Box
+          display={DISPLAY.FLEX}
+          marginTop={4}
+          marginRight={4}
+          marginBottom={4}
+          marginLeft={4}
+          paddingTop={3}
+          paddingRight={3}
+          paddingBottom={4}
+          paddingLeft={3}
+          borderRadius={BorderRadius.MD}
+          alignItems={AlignItems.center}
+          className="token-allowance-container__custom-nonce-content"
+        >
+          <Box
+            className="token-allowance-container__custom-nonce-header"
+            justifyContent={JustifyContent.flexStart}
+            alignItems={AlignItems.center}
+          >
+            <Text
+              variant={TextVariant.bodySm}
+              fontWeight={FONT_WEIGHT.NORMAL}
+              as="h6"
+            >
+              {t('nonce')}
+            </Text>
+            <Button
+              type="link"
+              className="token-allowance-container__custom-nonce-edit"
+              onClick={() =>
+                showCustomizeNonceModal({
+                  nextNonce,
+                  customNonceValue,
+                  updateCustomNonce,
+                  getNextNonce,
+                })
+              }
+            >
+              {t('edit')}
+            </Button>
+          </Box>
+          <Text
+            className="confirm-approve-content__custom-nonce-value"
+            variant={TextVariant.bodySmBold}
+            as="h6"
+          >
+            {customNonceValue || nextNonce}
+          </Text>
+        </Box>
+      )}
```

Comment:
It will look cleaner if we extract custom nonce code into a separate child component.

Reply:
@jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/17945/commits/cb62dbcddb0b653bb306db0c89f7e9bc72cd4284).